xdsl/interactive/app.py

```diff
@@ -249,10 +251,37 @@ def compute_available_pass_list(self) -> tuple[AvailablePass, ...]:
             case Exception():
                 return ()
             case ModuleOp():
+                # transform rewrites into passes
+                rewrites = get_all_possible_rewrites(
+                    self.current_module,
+                    individual_rewrite.REWRITE_BY_NAMES,
+                )
+                rewrites_as_pass_list: tuple[AvailablePass, ...] = ()
+                for op_idx, (op_name, pat_name) in rewrites:
+                    rewrite_pass = individual_rewrite.IndividualRewrite
+                    rewrite_spec_arg_str = f'matched_operation_index={op_idx} operation_name="{op_name}" pattern_name={pat_name}'
+                    rewrite_spec = list(
+                        parse_pipeline(f"{rewrite_pass.name}{{{rewrite_spec_arg_str}}}")
+                    )[0]
```

Comment:
It feels like here we have all the information to construct the spec programmatically, what's the advantage to parsing it? Constructing a list and getting the first element seems off

Reply:
You're right. changed.